encrypted-media/polyfill/edge-polyfill.js

```diff
@@ -0,0 +1,193 @@
+        }
+
+        MediaKeySession.prototype = Object.create( EventTarget.prototype );
+
+        Object.defineProperties( MediaKeySession.prototype, {
+            sessionId:  { get: function() { return this._sessionId ? this._sessionId : this._session.sessionId; } },
+            expiration: { get: function() { return this._session.expiration; } },
+            closed:     { get: function() { return this._closed; } },
+            keyStatuses:{ get: function() { return this._session.keyStatuses; } }
+        });
+
+        // load()
+        //
+        // Use a surrogate 'persistent-release-message' session to obtain the release message
+        //
+        MediaKeySession.prototype.load = function load( sessionId )
+        {
+            if ( this.sessionId ) return Promise.reject( new DOMException('InvalidAccessError') );
+
+            this._surrogate = this._mediaKeys.createSession( 'persistent-release-message' );
+            this._surrogate.addEventListener( 'message', this.dispatchEvent.bind( this ) );
+
+            return this._surrogate.load( sessionId ).then( function( success ) {
+                if (!success) return false;
+
+                this._sessionId = sessionId;
+                this._removing = true;
+                this._session.close();
+
+                return this._surrogate.remove().then( function() { return true; } );
+            }.bind( this ) );
+        };
+
+        // remove()
+        //
+        // On an existing session, use a surrogate 'persistent-release-message' session to obtain the release message
+        //
+        MediaKeySession.prototype.remove = function remove()
+        {
+            if ( this._sessionId !== undefined ) return Promise.reject( new DOMException('InvalidAccessError') );
+            if ( this.sessionId === undefined ) return Promise.reject( new DOMException('InvalidAccessError') );
+
+            this._surrogate = this._mediaKeys.createSession( 'persistent-release-message' );
+            this._surrogate.addEventListener( 'message', this.dispatchEvent.bind( this ) );
+            this._removing = true;
+            this._sessionId = this._session.sessionId;
+
+            var self = this;
+
+            return Promise.all( [ self._session.close(), self._session.closed ] ).then( function() {
+                return self._surrogate.load( self._sessionId );
+            }).then( function( success ) {
+                if ( !success ) {
+                    throw new DOMException('InvalidAccessError');
+                }
+
+                return self._surrogate.remove();
+            }).then( function() { return true; } );
+        }
+
+        // update()
+        //
+        // For a normal session, pass through, otherwise update the surrogate and close the proxy
+        MediaKeySession.prototype.update = function update( message )
+        {
+            if ( !this._removing ) return this._session.update( message );
```

Comment:
nit:  style inconsistencies (for example, “if ( !this._removing )” here and and “if (!this._removing)” on line 106 below).